src/Microsoft.IdentityModel.JsonWebTokens/Json/JsonClaimSet.cs

```diff
@@ -109,6 +109,10 @@ internal static void CreateClaimFromObject(List<Claim> claims, string claimType,
                     if (claim != null)
                         claims.Add(claim);
                 }
+            else
+            {
+                claims.Add(new Claim(claimType, value.ToString(), ClaimValueTypes.String, issuer, issuer));
```

Comment:
Above we only add claims to claims identity for known types. So with new read code, if custom claim class is added, it won't be added to the ClaimsIdentity. This line makes sure it does get added as a string. 

The reason why this line wasn't needed before is because we would end up [reading the claims with unknown names as strings](https://github.com/AzureAD/azure-activedirectory-identitymodel-extensions-for-dotnet/blob/faa4998627289e752124c0f5ceb6e0c4c453bc18/src/Microsoft.IdentityModel.Tokens/Json/JsonSerializerPrimitives.cs#L953).